src/common/io.py

```diff
@@ -1547,11 +1580,18 @@ def simulation_start(self):
                     cont_valref_bool.append(var.value_reference)
                     cont_name_bool.append(var.name)
                     cont_alias_bool.append(-1 if var.alias == fmi.FMI_NEGATED_ALIAS else 1)
-        
+
         # Open file
-        f = codecs.open(self.file_name,'w','utf-8')
-        self.file_open = True
-        
+        if isinstance(self.file_name, str):
+            f = codecs.open(self.file_name,'w','utf-8')
+            self.file_open = True
+        else:
+            if not hasattr(self.file_name, 'write'):
+                raise fmi.FMUException("Instance set via option 'result_file_name' needs to support 'write' as required.")
```

Comment:
Not sure that I understand the exception here. What about something like "Failed to write the result file. The provided "result_file_name" needs to be a filename or a class that supports writing to through the 'write' method."

Reply:
Fixed